test/sslapitest.c

```diff
@@ -225,133 +190,103 @@ static int test_keylog_output(char *buffer, const SSL *ssl,
                 return 0;
-            }
 
-            token = strtok(NULL, " \n");
-            if (!token) {
-                printf("Unexpected short client handshake secret log.\n");
+            if (!TEST_ptr(token = strtok(NULL, " \n")))
                 return 0;
-            }
-            if (strlen(token) != 64) {
-                printf("Bad value for client random: %s\n", token);
+            if (!TEST_size_t_eq(strlen(token), 64))
                 return 0;
-            }
-            if (compare_hex_encoded_buffer(token, 64, actual_client_random,
-                                           client_random_size)) {
-                printf("Bad value for client random: %s\n", token);
+            if (!TEST_false(compare_hex_encoded_buffer(token, 64,
+                                           actual_client_random,
+                                           client_random_size)))
                 return 0;
-            }
 
-            token = strtok(NULL, " \n");
-            if (!token) {
-                printf("Unexpectedly short master secret log.\n");
+            if (!TEST_ptr(token = strtok(NULL, " \n")))
                 return 0;
-            }
 
             /*
              * TODO(TLS1.3): test that application traffic secrets are what
              * we expect */
         } else {
-            printf("Unexpected token in buffer: %s\n", token);
+            TEST_info("Unexpected token %s\n", token);
             return 0;
         }
 
         token = strtok(NULL, " \n");
     }
 
-    /* Return whether we got what we expected. */
-    return ((rsa_key_exchange_count == expected->rsa_key_exchange_count) &&
-            (master_secret_count == expected->master_secret_count) &&
-            (client_handshake_secret_count == expected->client_handshake_secret_count) &&
-            (server_handshake_secret_count == expected->server_handshake_secret_count) &&
-            (client_application_secret_count == expected->client_application_secret_count) &&
-            (server_application_secret_count == expected->server_application_secret_count));
+    /* Got what we expected? */
+    if (!TEST_size_t_eq(rsa_key_exchange_count,
+                        expected->rsa_key_exchange_count)
+            || !TEST_size_t_eq(master_secret_count,
+                               expected->master_secret_count)
+            || !TEST_size_t_eq(client_handshake_secret_count,
+                               expected->client_handshake_secret_count)
+            || !TEST_size_t_eq(server_handshake_secret_count,
+                               expected->server_handshake_secret_count)
+            || !TEST_size_t_eq(client_application_secret_count,
+                               expected->client_application_secret_count)
+            || !TEST_size_t_eq(server_application_secret_count,
+                               expected->server_application_secret_count))
+        return 0;
+    return 1;
```

Comment:
the point is that it follows everything else "if test {||test...| fails, return 0;" :)

Everyone's pedanticism is probably better used elsewhere ::)